src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -111,7 +111,29 @@ std::shared_ptr<MulticlassLabels> CARTree::apply_multiclass(std::shared_ptr<Feat
 	auto current=get_root()->as<bnode_t>();
 
 	require(current, "Tree machine not yet trained.");
-	return apply_from_current_node(data->as<DenseFeatures<float64_t>>(), current)->as<MulticlassLabels>();
+	if (auto subfeat_data =
+	        std::dynamic_pointer_cast<DenseSubSamplesFeatures<float64_t>>(data))
+	{
+		auto num_feat = subfeat_data->get_dim_feature_space();
+		auto num_vec = subfeat_data->get_num_vectors();
+		SGMatrix<float64_t> feature_matrix(num_feat, num_vec);
```

Comment:
actually, the is commodity code, it should be somewhere where it can be re-used, for example in the form of a copy constructor, or a method that returns the new feature object

Reply:
add `DenseFeatures::DenseFeatures(std::shared_ptr<DenseSubSamplesFeatures<T>>& data);` ?